common/tvm_utility/README.md

```diff
@@ -50,35 +50,17 @@ error description.
 
 ### Neural Networks Provider
 
-This package also provides a utility to get pre-compiled neural networks to packages using them for their inference.
-
 The neural networks are compiled as part of the
 [Model Zoo](https://github.com/autowarefoundation/modelzoo/) CI pipeline and saved to an S3 bucket.
-This package exports cmake variables and functions for ease of access to those neural networks.
 
 The `get_neural_network` function creates an abstraction for the artifact management.
-The artifacts are saved under the source directory of the package making use of the function; under "data/".
-Priority is given to user-provided files, under "data/user/${MODEL_NAME}/".
-If there are no user-provided files, the function tries to reuse previously-downloaded artifacts.
-If there are no previously-downloaded artifacts, and if the `DOWNLOAD_ARTIFACTS` cmake variable is set, they will be downloaded from the bucket.
-Otherwise, nothing happens.
+Users should provide model files under "data/user/${MODEL_NAME}/". Otherwise, nothing happens and compilation of the package will be skipped.
```

Comment:
If the artifacts are required for compilation, then they should be part of the source tree (via `git-lfs` for example if the files are too big), otherwise it won't be possible to build Debian packages.

Reply:
Actually It is a bit confusing for now. Because this packages don't seem to build anyway now because DOWNLOAD_ARTIFACTS flag is disabled by default and this PR kinda show it because last changes was made several months ago and bug was hidden. So here I just follow the same logic. 

And as I understand before we are going to use ansible to provide artifacts. And for packages with tenosorrt support It  is more or less straight forward process as all model conversion from onnx to TRT happens later on the first run not on the build. 

For TVM it is a bit more complicated as it provide models in already compiled form and to use it in the package you will need to provide some model files to build it.  

But if I understand you correctly here. You propose that there should be some default artifacts for all TVM packages which should be part of the source tree and the only way to do it is `git-lfs` because for now `deploy_param.params` file weights 10-20 Mb usually. And If user will want to provide his own models wouldn't it be messy a bit ?

I think there is a kinda similar way to complile the model to TVM before run by the user same as with TRT but I'm not so familiar with TVM.

And what is the idea behind the Debian packages? Should user be able to use them of the shelf with the build in model? I guess in many cases user with have to use model trained on his own data.

Reply:
> For TVM it is a bit more complicated as it provide models in already compiled form and to use it in the package you will need to provide some model files to build it.

As far as I know, we only need the headers for building the packages, not the full models.

> And what is the idea behind the Debian packages? Should user be able to use them of the shelf with the build in model? I guess in many cases user with have to use model trained on his own data.

The Debian packages are a way for users to install Autoware in production environments without having to build it themselves, much like how ROS is distributed. Eventually, Autoware can become part of the ROS distribution, but that will come at a later stage.